cmake/build_apk.cmake

```diff
@@ -411,22 +410,87 @@ function(add_shader_library target)
           endforeach()
         endif()
 
-        add_custom_command (
-          OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${rel_pos}.spv
-          WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-          COMMENT "Compiling SPIR-V binary ${shader}"
-          DEPENDS ${shader} ${FILE_DEPS}
-            ${VulkanTestApplications_SOURCE_DIR}/cmake/generate_cmake_dep.py
-          COMMAND ${CMAKE_GLSL_COMPILER} -mfmt=c -o ${output_file} -c ${temp} -MD
-            ${ADDITIONAL_ARGS}
-          COMMAND ${PYTHON_EXECUTABLE}
-            ${VulkanTestApplications_SOURCE_DIR}/cmake/generate_cmake_dep.py
-            ${output_file}.d
-          COMMAND ${CMAKE_COMMAND} -E
-            copy_if_different
-              ${output_file}.d.cmake.tmp
-              ${output_file}.d.cmake
-        )
+        # Compile HLSL shaders through glslc and DXC
+        get_filename_component(shader_name ${shader} NAME)
+
+        if (${shader_name} MATCHES ".*\\.hlsl\\..*")
+          message(STATUS "Found HLSL Shader: ${shader_name}")
```

Comment:
These are getting large enough that we should probably split them out into their own functions.